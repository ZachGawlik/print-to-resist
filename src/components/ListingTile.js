import React, { PropTypes } from 'react';
import '../styles/ListingTile.css';
import ColorIcon from './ColorIcon';
import ListingLink from './ListingLink';
import ListingThumbnail from './ListingThumbnail';

export default class ListingTile extends React.Component {
  static propTypes = {
    listing: PropTypes.object.isRequired
  };
  render() {
    const { listing } = this.props;
    return (
      <div className="ListingTile">
        <div className="ListingTile__card">
          <div className="ListingTile__content">
            <div className="ListingTile__headline">
              <p className="ListingTile__title overflow-ellipsis">
                <ListingLink listingId={listing.listingId}>
                  {listing.title}
                </ListingLink>
              </p>
              <ColorIcon isColor={listing.isColor} />
            </div>
            <div className="overflow-ellipsis">
              {listing.tags.length ?
                listing.tags.map(tag => (
                  <span className="ListingTile__tag" key={tag}>#{tag}</span>
                )) :
                <span className="ListingTile__tag--placeholder">no tags</span>
              }
            </div>
          </div>
          <ListingLink listingId={listing.listingId}>
            <ListingThumbnail
              title={listing.title}
              thumbnailId={listing.thumbnailId}
            />
          </ListingLink>
        </div>
      </div>
    );
  }
}
