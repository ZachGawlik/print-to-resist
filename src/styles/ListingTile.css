.ListingTile {
  padding: 10px;
  margin: 0 auto;
}

.ListingTile__card {
  box-shadow: 0 1px 3px 1px #ddd, 0 0 0 1px #ddd;
}

.ListingTile__content {
  padding: 10px;
}

.ListingTile__headline {
  display: flex;
  justify-content: space-between;
  align-items: baseline;
}

.ListingTile__title {
  margin: 5px 0;
  font-size: larger;
}

.ListingTile__deadline {
  color: grey;
  margin: 0;
}

.ListingTile__tag--placeholder {
  color: grey;
  font-style: italic;
}
