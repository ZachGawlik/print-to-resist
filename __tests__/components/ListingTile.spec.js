import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ListingTile from '../../src/components/ListingTile';

describe('ListingTile', () => {
  it('renders as expected', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <ListingTile
        tags={[]}
        title="Upcoming Town Halls"
        thumbnailId="testImageId"
      />
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
