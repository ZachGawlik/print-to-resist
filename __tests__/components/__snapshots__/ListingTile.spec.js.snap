// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingTile renders as expected 1`] = `
<div
  className="ListingTile"
>
  <div
    className="ListingTile__content"
  >
    <div
      className="ListingTile__headline"
    >
      <p
        className="ListingTile__title overflow-ellipsis"
      >
        Upcoming Town Halls
      </p>
      <p
        className="ListingTile__deadline"
      />
    </div>
    <div
      className="overflow-ellipsis"
    >
      <span
        className="ListingTile__tag--placeholder"
      >
        no tags
      </span>
    </div>
  </div>
  <div
    className="ListingTile__image-wrapper"
  >
    <img
      alt=""
      className="ListingTile__image"
      src="http://localhost:3000/image/testImageId"
    />
  </div>
</div>
`;
